tests/test_parquet.py

```diff
@@ -1511,5 +1572,142 @@ def testStorageClass(self):
 
 
+@unittest.skipUnless(np is not None, "Cannot test ParquetFormatterArrowNumpy without numpy.")
+@unittest.skipUnless(pa is not None, "Cannot test ParquetFormatterArrowNumpy without pyarrow.")
+class ParquetFormatterArrowNumpyDictTestCase(unittest.TestCase):
+    """Tests for ParquetFormatter, ArrowNumpyDict, using local file store."""
+
+    configFile = os.path.join(TESTDIR, "config/basic/butler.yaml")
+
+    def setUp(self):
+        """Create a new butler root for each test."""
+        self.root = makeTestTempDir(TESTDIR)
+        config = Config(self.configFile)
+        self.butler = Butler(Butler.makeRepo(self.root, config=config), writeable=True, run="test_run")
+        # No dimensions in dataset type so we don't have to worry about
+        # inserting dimension data or defining data IDs.
+        self.datasetType = DatasetType(
+            "data", dimensions=(), storageClass="ArrowNumpyDict", universe=self.butler.registry.dimensions
+        )
+        self.butler.registry.registerDatasetType(self.datasetType)
+
+    def tearDown(self):
+        removeTestTempDir(self.root)
+
+    def testNumpyDict(self):
+        tab1 = _makeSimpleNumpyTable(include_multidim=True)
+        dict1 = _numpy_to_numpy_dict(tab1)
+
+        self.butler.put(dict1, self.datasetType, dataId={})
+        # Read the whole table.
+        dict2 = self.butler.get(self.datasetType, dataId={})
+        self._checkNumpyDictEquality(dict1, dict2)
+        # Read the columns.
+        columns2 = self.butler.get(self.datasetType.componentTypeName("columns"), dataId={})
+        self.assertEqual(len(columns2), len(dict1.keys()))
+        for i, name in enumerate(dict1.keys()):
+            self.assertEqual(columns2[i], name)
+        # Read the rowcount.
+        rowcount = self.butler.get(self.datasetType.componentTypeName("rowcount"), dataId={})
+        self.assertEqual(rowcount, len(dict1["a"]))
+        # Read the schema.
+        schema = self.butler.get(self.datasetType.componentTypeName("schema"), dataId={})
+        self.assertEqual(schema, ArrowNumpySchema(tab1.dtype))
+        # Read just some columns a few different ways.
+        tab3 = self.butler.get(self.datasetType, dataId={}, parameters={"columns": ["a", "c"]})
+        subdict = {key: dict1[key] for key in ["a", "c"]}
+        self._checkNumpyDictEquality(subdict, tab3)
+        tab4 = self.butler.get(self.datasetType, dataId={}, parameters={"columns": "a"})
+        subdict = {key: dict1[key] for key in ["a"]}
+        self._checkNumpyDictEquality(subdict, tab4)
+        tab5 = self.butler.get(self.datasetType, dataId={}, parameters={"columns": ["index", "a"]})
+        subdict = {key: dict1[key] for key in ["index", "a"]}
+        self._checkNumpyDictEquality(subdict, tab5)
+        tab6 = self.butler.get(self.datasetType, dataId={}, parameters={"columns": "ddd"})
+        subdict = {key: dict1[key] for key in ["ddd"]}
+        self._checkNumpyDictEquality(subdict, tab6)
```

Comment:
These tests were put in because of a previous bug where a single column was being read as a string array.  So `"ddd"` checks that it gets the `"ddd"` column back and not the `"d"` column (de-duplicated).  And I just copied that test everywhere even though it may only need to be in one place.